Mathlib/Data/Matroid/Basic.lean

```diff
@@ -566,16 +566,15 @@ theorem Base.eq_of_subset_indep (hB : M.Base B) (hI : M.Indep I) (hBI : B ⊆ I)
   let ⟨B', hB', hB'I⟩ := hI.exists_base_superset
   hBI.antisymm (by rwa [hB.eq_of_subset_base hB' (hBI.trans hB'I)])
 
-theorem base_iff_maximal_indep : M.Base B ↔ M.Indep B ∧ ∀ I, M.Indep I → B ⊆ I → B = I := by
-  refine ⟨fun h ↦ ⟨h.indep, fun _ ↦ h.eq_of_subset_indep ⟩, fun ⟨h, h'⟩ ↦ ?_⟩
+theorem base_iff_maximal_indep : M.Base B ↔ Maximal M.Indep B := by
+  rw [maximal_subset_iff]
+  refine ⟨fun h ↦ ⟨h.indep, fun _ ↦ h.eq_of_subset_indep⟩, fun ⟨h, h'⟩ ↦ ?_⟩
```

Comment:
When doing a refactor, adding unnecessary changes is *not* helping. I understand that your PR is motivated by applications in matroid theory, but I don't need to see the changes in an already big diff to be convinced.

You have already added them here, so I won't ask you to revert, but this is typically the kind of changes that can/should be left to a separate PR.